server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -236,11 +238,24 @@ public static class FileSettingsHealthIndicatorService implements HealthIndicato
             )
         );
 
+        public static final String DESCRIPTION_LENGTH_LIMIT_KEY = "fileSettings.descriptionLengthLimit";
+        static final Setting<Integer> DESCRIPTION_LENGTH_LIMIT = Setting.intSetting(
+            DESCRIPTION_LENGTH_LIMIT_KEY,
+            100,
+            1, // Need room for the ellipsis
```

Comment:
If someone sets this to the max indexable field limit, then the ellipsis will push this over the edge, which might be surprising to discover at runtime.  Should we account for that here by setting a max that references this value?  Maybe a note in the javadoc here is good enough, IDK.

Reply:
I don't quite understand... the shortening logic never makes the string longer.